src/corespotlight.cs

```diff
@@ -1105,4 +1117,123 @@ interface CSImportExtension : NSExtensionRequestHandling
 		bool Update (CSSearchableItemAttributeSet attributes, NSUrl contentUrl, [NullAllowed] out NSError error);
 	}
 
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSSearchQuery))]
+	[DisableDefaultCtor]
+	interface CSUserQuery
+	{
+		[Export ("initWithUserQueryString:queryContext:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string userQueryString, CSUserQueryContext queryContext);
+
+		[Export ("foundSuggestionCount")]
+		nint FoundSuggestionCount { get; }
+
+		[NullAllowed, Export ("foundSuggestionsHandler", ArgumentSemantic.Copy)]
+		Action<NSArray<CSSuggestion>> FoundSuggestionsHandler { get; set; }
+
+		[Export ("start")]
+		void Start ();
+
+		[Export ("cancel")]
+		void Cancel ();
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSSearchQueryContext))]
+	interface CSUserQueryContext
+	{
+		[Static]
+		[Export ("userQueryContext")]
+		CSUserQueryContext UserQueryContext { get; }
+
+		[Static]
+		[Export ("userQueryContextWithCurrentSuggestion:")]
+		CSUserQueryContext Create ([NullAllowed] CSSuggestion currentSuggestion);
+
+		[Export ("maxSuggestionCount")]
+		nint MaxSuggestionCount { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSUserQueryContext))]
+	interface CSTopHitQueryContext
+	{
+		[Static]
+		[Export ("topHitQueryContext")]
+		CSTopHitQueryContext TopHitQueryContext { get; }
+
+		[Static]
+		[Export ("topHitQueryContextWithCurrentSuggestion:")]
+		CSTopHitQueryContext Create ([NullAllowed] CSSuggestion currentSuggestion);
+
+		[Export ("maxItemCount")]
+		nint MaxItemCount { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof(CSUserQuery))]
+	[DisableDefaultCtor]
+	interface CSTopHitQuery
+	{
+		[Export ("initWithUserQueryString:queryContext:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string userQueryString, CSTopHitQueryContext queryContext);
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	interface CSSuggestion : NSSecureCoding, NSCopying
```

Comment:
This seems to be used as a "Result" object and it is readonly

```suggestion
	[DisableDefaultCtor]
	interface CSSuggestion : NSSecureCoding, NSCopying
```